common/meta/types/src/user_stage.rs

```diff
@@ -227,18 +229,29 @@ pub struct UserStageInfo {
     pub file_format_options: FileFormatOptions,
     pub copy_options: CopyOptions,
     pub comment: String,
+    pub number_of_files: u64,
+    pub creator: Option<UserIdentity>,
+    pub created_on: DateTime<Utc>,
+    pub updated_on: DateTime<Utc>,
 }
 
-impl TryFrom<Vec<u8>> for UserStageInfo {
-    type Error = ErrorCode;
-
-    fn try_from(value: Vec<u8>) -> Result<Self> {
-        match serde_json::from_slice(&value) {
-            Ok(info) => Ok(info),
-            Err(serialize_error) => Err(ErrorCode::IllegalUserStageFormat(format!(
-                "Cannot deserialize stage from bytes. cause {}",
-                serialize_error
-            ))),
+impl UserStageInfo {
+    pub fn get_prefix(&self) -> String {
+        match self.stage_type {
+            StageType::External => "/".to_string(),
+            StageType::Internal => {
+                // It's internal, so we should prefix with stage name.
+                format!("/stage/{}/", self.stage_name)
+            }
         }
     }
 }
+
+#[derive(Default, Clone)]
+pub struct StageFile {
+    pub name: String,
```

Comment:
How about using `path` to make the behavior clear?

Reply:
SGTM.